src/esp/assets/MeshMetaData.h

```diff
@@ -5,10 +5,46 @@
 #pragma once
 
 #include "esp/core/esp.h"
+#include "esp/gfx/magnum.h"
 
 namespace esp {
 namespace assets {
 
+//! Some mesh files include a transformation hierarchy. A @ref MeshTransformNode
+//! stores this hierarchy and indices for the meshes at each level such that it
+//! can be reused to instances meshes later.
+struct MeshTransformNode {
+  //! mesh ID within @ref MeshMetaData::meshIndex
+  int meshIDLocal;
+
+  //! material ID within @ref MeshMetaData::materialIndex
+  int materialIDLocal;
+
+  //! Object ID in the original file
+  int componentID;
+
+  std::vector<MeshTransformNode> children;
+
+  //! Node local transform
+  Magnum::Matrix4 T;
+
+  MeshTransformNode() {
+    meshIDLocal = ID_UNDEFINED;
+    materialIDLocal = ID_UNDEFINED;
+    componentID = ID_UNDEFINED;
+  };
+
+  MeshTransformNode(const MeshTransformNode& val) {
+    componentID = val.componentID;
+    meshIDLocal = val.meshIDLocal;
+    materialIDLocal = val.materialIDLocal;
+    T = Magnum::Matrix4(val.T);
+    for (auto& child : val.children) {
+      children.push_back(MeshTransformNode(child));
+    }
+  }
+};
```

Comment:
This is just a copy constructor, right? Why not let the compiler generate it automatically?